core/conversion/converters/impl/element_wise.cpp

```diff
@@ -14,237 +14,267 @@ nvinfer1::ILayer* add_elementwise(
-    auto self_shuffle = ctx->net->addShuffle(*self);
-    self_shuffle->setReshapeDimensions(util::toDimsPad(self_dims_vec, other_dims_vec.size()));
-    self_shuffle->setName(
-        std::string("[Reshape self to " + util::toStr(self_dims) + " for broadcasting (" + name + ")]").c_str());
-    self = self_shuffle->getOutput(0);
+    const std::string& name) {
+  // ensure self to have larger number of dimension
+  bool swapSelfOther = false;
+  if (self->getDimensions().nbDims < other->getDimensions().nbDims) {
+    std::swap(self, other);
+    swapSelfOther = true;
   }
-
-  nvinfer1::ILayer* ele;
-  if (scalar != 1) {
-    LOG_WARNING("Please verify scalar handling in add converter, channel axis set to 3 but scaling is uniform");
-
-    auto shape = util::toVec(other_dims);
-
-    if (shape.size() < 4) {
-      auto new_shape = util::toDimsPad(shape, 4);
-      LOG_DEBUG(
-          "Input shape is less than 4D got: "
-          << util::toDims(shape) << ", inserting shuffle layers to reshape to 4D tensor shape: " << new_shape);
-      auto other_shuffle = ctx->net->addShuffle(*other);
-      other_shuffle->setReshapeDimensions(new_shape);
-      other_shuffle->setName(std::string("[Reshape other to " + util::toStr(new_shape) + ']').c_str());
-      other = other_shuffle->getOutput(0);
-
-      auto self_shuffle = ctx->net->addShuffle(*self);
-      self_shuffle->setReshapeDimensions(new_shape);
-      self_shuffle->setName(std::string("[Reshape self to " + util::toStr(new_shape) + ']').c_str());
-      self = self_shuffle->getOutput(0);
+  auto selfDim = util::toVec(self->getDimensions());
+  auto otherDim = util::toVec(other->getDimensions());
+  if (selfDim.size() != otherDim.size()) {
+    // other is with dynamic shape, need to expand its dimension now and get its shape at runtime
+    if (otherDim.end() != std::find(otherDim.begin(), otherDim.end(), -1)) {
+      auto thOtherStaticShapeMask = torch::ones(selfDim.size(), torch::kInt32);
+      auto thOtherDynamicShapeMask = torch::zeros(selfDim.size(), torch::kInt32);
+      for (size_t start = selfDim.size() - otherDim.size(), idx = 0; idx < otherDim.size(); ++idx) {
+        if (-1 != otherDim[idx]) {
+          thOtherStaticShapeMask[start + idx] = otherDim[idx];
+        } else {
+          thOtherStaticShapeMask[start + idx] = 0;
+          thOtherDynamicShapeMask[start + idx] = 1;
+        }
+      }
+      auto otherStaticShapeMask = tensor_to_const(ctx, thOtherStaticShapeMask);
+      auto otherDynamicShapeMask = tensor_to_const(ctx, thOtherDynamicShapeMask);
+      auto selfShape = ctx->net->addShape(*self)->getOutput(0);
+      // size of dynamic dimension of other need to the same as that of corresponding dimension of self
+      auto otherDynamicShape =
+          ctx->net->addElementWise(*selfShape, *otherDynamicShapeMask, nvinfer1::ElementWiseOperation::kPROD)
+              ->getOutput(0);
+      auto targetOhterShape =
```

Comment:
targetOhterShape -> targetOtherShape  (minor correction in variable name)

Reply:
updated